Source/IntegrationTests/LitTests.cs

```diff
@@ -0,0 +1,98 @@
+      "/errorTrace:0",
+      
+      // Set a default time limit, to catch cases where verification time runs off the rails
+      "/timeLimit:300"
+    };
+
+    private static ILitCommand MainWithArguments(Assembly assembly, IEnumerable<string> arguments, LitTestConfiguration config, bool invokeDirectly) {
+      return MainMethodLitCommand.Parse(assembly, arguments, config, invokeDirectly);
+    }
+
+    private static readonly LitTestConfiguration CONFIG = new() {
+      Commands = new Dictionary<string, Func<IEnumerable<string>, LitTestConfiguration, ILitCommand>> {
+        { "%baredafny", (args, config) =>
+          MainWithArguments(dafnyDriverAssembly, args, config, false) },
+        { "%dafny", (args, config) =>
+          MainWithArguments(dafnyDriverAssembly, defaultDafnyArguments.Concat(args), config, false) },
+        { "%server", (args, config) =>
+          MainWithArguments(dafnyServerAssembly, args, config, false) },
+      },
+
+      Substitions = new Dictionary<string, string> {
+        { "%diff", "diff" },
+        { "%binaryDir", "." },
+        { "%z3", Path.Join("z3", "bin", "z3") },
+        { "%refmanexamples", Path.Join("TestFiles", "LitTests", "LitTest", "refman", "examples") }
+      },
+
+      PassthroughEnvironmentVariables = new[] { "PATH", "HOME" },
+    };
+
+    static LitTests() {
+      var dafnyReleaseDir = Environment.GetEnvironmentVariable("DAFNY_RELEASE");
+      if (dafnyReleaseDir != null) {
+        CONFIG.Commands["%baredafny"] = (args, config) =>
+          new ShellLitCommand(config, Path.Join(dafnyReleaseDir, "dafny"), args, config.PassthroughEnvironmentVariables);
+        CONFIG.Commands["%dafny"] = (args, config) =>
+          new ShellLitCommand(config, Path.Join(dafnyReleaseDir, "dafny"), defaultDafnyArguments.Concat(args), config.PassthroughEnvironmentVariables);
+        CONFIG.Commands["%server"] = (args, config) =>
+          new ShellLitCommand(config, Path.Join(dafnyReleaseDir, "DafnyServer"), args, config.PassthroughEnvironmentVariables);
+        CONFIG.Substitions["%z3"] = Path.Join(dafnyReleaseDir, "z3", "bin", "z3");
+      }
+
+      if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) {
+        CONFIG.Features = new[] { "ubuntu", "posix" };
+      } else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+        CONFIG.Features = new[] { "windows" };
+      } else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) {
+        CONFIG.Features = new[] { "macosx", "posix" };
+      } else {
+        throw new Exception($"Unsupported OS: {RuntimeInformation.OSDescription}");
+      }
+    }
+
+    private readonly ITestOutputHelper output;
+
+    public LitTests(ITestOutputHelper output) {
+      this.output = output;
+    }
+
+    [FileTheory]
+    [FileData(Includes = new[] { "**/*.dfy", "**/*.transcript" },
+              Excludes = new[] { "**/Inputs/**/*", "refman/examples/**/*" })]
```

Comment:
I had a few output files from a previous run that were accidentally picked up as Lit tests, for example this one:

`TestFiles/LitTests/LitTest/allocated1/dafny0/Output/Calculations.dfy.tmp.dprint.dfy`

Reply:
Right, lit automatically filters out Output files internally. I've addressed this by adding another `Exclude` here even though the lit test runner should really be responsible for that, since it's a fair bit easier.

Ideally it would be automatic, but another improvement I want to make after this PR is to run each test in a fresh temporary directory instead of a fixed and shared `Output` directory (I started doing that in earlier versions of this PR in fact but have since trimmed that scope for v1), which will avoid potentially polluting results and also make this issue moot.